src/window_type.h

```diff
@@ -10,6 +10,9 @@
 #ifndef WINDOW_TYPE_H
 #define WINDOW_TYPE_H
 
+/** Widget ID. */
+using WidgetID = int;
```

Comment:
Maybe make it clear that negative WidgetIDs are special.
```suggestion
/**
 * Widget ID.
 * Even though the ID is signed, actual IDs must be non-negative.
 * Negative IDs are used for special cases, like denoting 'no widget'. 
 */
using WidgetID = int;
```